images.go

```diff
@@ -528,6 +576,28 @@ func (r *imageStore) SetBigData(id, key string, data []byte) error {
 			image.BigDataNames = append(image.BigDataNames, key)
 			save = true
 		}
+		if key == ImageDigestBigDataKey {
+			if oldDigest != "" && oldDigest != newDigest {
+				// remove the image from the list of images in the digest-based
+				// index which corresponds to the old digest for this item
+				if list, ok := r.bydigest[oldDigest]; ok {
+					prunedList := imageSliceWithoutValue(list, image)
+					if len(prunedList) == 0 {
+						delete(r.bydigest, oldDigest)
```

Comment:
If you delete thie r.bydigest, will the 

 +			list := r.bydigest[newDigest] 

Still work?

Reply:
Yes, it returns the zero value, which is `nil`.